src/xrdreq/QservRequest.cc

```diff
@@ -161,6 +179,7 @@ void QservRequest::ProcessResponseData(const XrdSsiErrInfo& eInfo, char* buff, i
             //          downstream.
             proto::FrameBufferView view(_buf, _bufSize);
             onResponse(view);
+            _self = nullptr;
```

Comment:
Good idea ... except the right way of doing this is:
```
auto localSelfKeepAlive = std::move(_self);
```
Otherwise, `_self` will never get deleted if any crash happens before `_self = nullptr;`

Reply:
It depends. If you set `localSelf` at the start, you can set `_self = nullptr;` at any time following and it will behave properly and delete `_self` when the function exits. If you don't set `_self` to nullptr, the `localSelf` has little effect. I mention this as `_self`  is set to nullptr a few times, and everytime that happens that block of code must end with some form of `return`. I just think setting localSelf at the start my be less prone to errors, especially a couple of years down the line.

If the program crashes, a memory leak isn't going to make much difference. Either method is fine. The move is faster but requires more conscious effort.

Reply:
Regarding your statement that:

> If the program crashes, a memory leak isn't going to make much difference. Either method is fine. The move is faster but requires more conscious effort.

That's not always the case here if the code throws an exception that's properly handled. Here is how it would be addressed:
```
{
    auto self = std::move(_self);
    proto::FrameBufferView view(_buf, _bufSize);   // <- may throw
    onResponse(view);    // <- may throw
    // self gets destroyed and the reference counter gets decremented regardless of any exceptions thrown
}
```

Reply:
If you look at the design of the base class `xrdreq:: QservRequest` and its derived classes then it would be clear why the reference counter (via `_self = nullptr;`, etc.) gets decremented after sending an upstream notification to the derived classes by calling either of two virtual methods `onFailed` or `onResponse`. After that, the object is useless and can be safely eliminated w/o leaving any side effects.

Making a copy of the shared pointer at the start of `QservRequest::ProcessResponseData()` doesn't help to achieve this goal.